components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/ApplicationImportExportManager.java

```diff
@@ -134,28 +134,56 @@ public boolean isOwnerAvailable(String userId) throws APIManagementException {
+                            // add subscription if update flag is not specified
+                            // it will throw an error if subscriber already exists
+                            if (update == null || !update) {
+                                apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
+                            } else if (!apiConsumer.isSubscribedToApp(subscribedAPI.getApiId(), userId, appId)) {
+                                // on update skip subscriptions that already exists
+                                apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
+                            }
+                        } else {
+                            log.error("Failed to import Subscription as API " + name + "-" + version +
+                                    " as one or more tiers may be unavailable or the API may not have been published ");
+                            skippedAPIList.add(subscribedAPI.getApiId());
                         }
                     } else {
-                        log.error("Failed to import Subscription as API " + name + "-" + version +
-                                " as one or more tiers may be unavailable or the API may not have been published ");
-                        skippedAPIList.add(subscribedAPI.getApiId());
+                        //Handle API subscriptions
+                        API api = (API) apiSet.iterator().next();
+                        //tier of the imported subscription
+                        Tier tier = subscribedAPI.getTier();
+                        //checking whether the target tier is available
+                        if (isTierAvailable(tier, api) && api.getStatus() != null &&
+                                APIConstants.PUBLISHED.equals(api.getStatus())) {
+                            ApiTypeWrapper apiTypeWrapper = new ApiTypeWrapper(api);
+                            apiTypeWrapper.setTier(tier.getName());
+                            // add subscription if update flag is not specified
+                            // it will throw an error if subscriber already exists
+                            if (update == null || !update) {
+                                apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
+                            } else if (!apiConsumer.isSubscribedToApp(subscribedAPI.getApiId(), userId, appId)) {
+                                // on update skip subscriptions that already exists
+                                apiConsumer.addSubscription(apiTypeWrapper, userId, appId);
+                            }
+                        } else {
+                            log.error("Failed to import Subscription as API " + name + "-" + version +
+                                    " as one or more tiers may be unavailable or the API may not have been published ");
+                            skippedAPIList.add(subscribedAPI.getApiId());
+                        }
```

Comment:
I think we can remove this whole else part.

Reply:
Fixed